stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/WebSocketUtil.java

```diff
@@ -575,6 +576,28 @@ public static void countDownForHandshake(ObjectValue webSocketClient) {
         }
     }
 
+    public static WebSocketException getWebSocketException(String msg, Throwable error, String errorCode,
+                                                           ErrorValue cause) {
+        WebSocketException exception;
+        String message = errorCode.substring(2) + ": " + msg;
+        if (!msg.isEmpty()) {
+            exception = new WebSocketException(message);
+        } else if (error != null) {
+            exception = new WebSocketException(error);
+        } else {
+            exception = new WebSocketException(message, cause);
+        }
+        BallerinaErrors.setTypeId(errorCode, WebSocketConstants.PROTOCOL_HTTP_PKG_ID, exception);
+        return exception;
+    }
+
+    public static void setNotifyFailure(String msg, NonBlockingCallback callback) {
+        WebSocketException exception = new WebSocketException(msg);
+        BallerinaErrors.setTypeId(WebSocketConstants.ErrorCode.WsInvalidHandshakeError.toString(),
```

Comment:
Unless we trap this error in Ballerina and convert to a error value, I don't think we need to do this here.

Reply:
Had a discussion with @kalaiyarasiganeshalingam and it turned out this exception is mapped to Ballerina distinct error defined here https://github.com/ballerina-platform/ballerina-lang/pull/23896/files#diff-0dc26ba9e7a9ac8f581ee802d2a2338dR21
Hence this change is needed.